### PR TITLE
Modify API fitting tutorial

### PR DESCRIPTION
**Pull request by Astro-Kirsty**
This PR is to address https://github.com/gammapy/gammapy/issues/5291#issuecomment-2163313687

We want to avoid accessing `datasets.parameters[name]` but rather take directly from the model

### CODE REVIEW

**Comment by Astro-Kirsty** [comment]
I am also going to adjust the plotting of the `stat_contour` because these lines can be confusing for the user

**Comment by Astro-Kirsty** [comment]
https://astro-kirsty.github.io/gammapy-docs-preview/tutorials/api/fitting.html#sphx-glr-tutorials-api-fitting-py

**Comment by bkhelifi** [comment]
I have a side question: the discusssion outcomes was not to use the FitResult object within the tutorials?

**Comment by Astro-Kirsty** [comment]
I thought the end decision was to use `datasets.models` or `fit_result.models`

**Comment by AtreyeeS** [comment]
> I have a side question: the discusssion outcomes was not to use the FitResult object within the tutorials?

@bkhelifi : Thanks to @QRemy recent efforts, `datasets.models` is now complete.
We can, of course, expose better `fit_results` as a matter of habit, but it is not urgent

**Review by AtreyeeS — COMMENTED** [review]
Thank you @Astro-Kirsty !

While having the stat_contours in a loop looks impressive, the feedback that I got is that because it loops over the parameters, users find it rather difficult to understand how to compute it for some specific parameter and not all combinations of all for parameters.

I would say we show something like

```
par1  = crab_model.parameters["alpha"]
par2 = ...
contour = fit.stat_contour(
                 datasets=datasets,
                 x=par1,
                 y=par2,
                 numpoints=11,
                 sigma=1,
             )
```

then explain what `contour` contains and show how to plot it using `plot_contour_line`.

Similarly for the `stat_surface` and the 1d `stat_profile`

File: examples/tutorials/api/fitting.py [hunk 1]
```diff
@@ -357,69 +358,52 @@ def make_contours(fit, datasets, result, npoints, sigmas):
...
-        "cx": (lambda ct: ct["contour_alpha_beta"]["alpha"]),
-        "cy": (lambda ct: ct["contour_alpha_beta"]["beta"]),
-    },
-]
-
 
-######################################################################
-# Finally we produce the confidence contours figures.
+#####################################################################
+# Produce the confidence contours figures.
```
Comment (QRemy) [inline examples/tutorials/api/fitting.py:388]: This could also go in a function that take `contours`, `params`, `sigmas` as input.
(We could even have that somewhere in gammapy together with the make_contours function but that would be for another PR)

**Review by QRemy — APPROVED** [review]
Thanks @Astro-Kirsty looks good (especially the simplification  of the ugly plotting code that I may have written).
